java/vector/src/main/java/org/apache/arrow/vector/file/ArrowReader.java

```diff
@@ -89,7 +89,8 @@ public Dictionary lookup(long id) {
     }
   }
 
-  public void loadNextBatch() throws IOException {
+  // Returns true if a batch was read, false on EOS
+  public boolean loadNextBatch() throws IOException {
```

Comment:
We need a better way to signal EOS than a length-0 record batch (which is ambiguous)